ranch/priority.go

```diff
@@ -91,21 +93,20 @@ func newRequestQueue() *requestQueue {
 
 // update updates expiration time is updated if already present,
 // add a new requestID at the end otherwise (FIFO)
-func (rq *requestQueue) update(requestID string, newExpiration time.Time) bool {
-
+func (rq *requestQueue) update(requestID string, newExpiration time.Time) (bool, time.Time) {
```

Comment:
rather than plumbing the creation time back through all of these methods, it might be better to create another getter method which can fetch the creation time using the request id.

Reply:
Done.